Include/cpython/pystate.h

```diff
@@ -106,6 +106,7 @@ struct _ts {
 
     PyObject *async_exc; /* Asynchronous exception to raise */
     unsigned long thread_id; /* Thread id where this tstate was created */
+    unsigned long native_thread_id; /* Native thread id, when available */
```

Comment:
Please document the behavior when PyThread_get_thread_native_id() is not available (field set to 0 if I understand correctly). How can a dev know if the function is available? Is it ok to check if the PY_HAVE_THREAD_NATIVE_ID macro is defined?

The comment doesn't mention "where this tstate was created".